arch/arm/core/aarch32/irq_relay.S

```diff
@@ -104,5 +104,215 @@ SECTION_FUNC(vector_relay_table, __vector_relay_table)
 	.word __vector_relay_handler
 	.word __vector_relay_handler
 	.word __vector_relay_handler
+#ifdef CONFIG_ARMV7_M_ARMV8_M_MAINLINE
+	.word __vector_relay_handler
```

Comment:
Since not all 240 external interrupts are used on all MCUs, I'm wondering if it is possible to reduce the number of defined vectors to conserve flash memory.

Reply:
@anangl Is interrupts number available via DT somehow?

Reply:
Yes, they are.
@nvlsianpu I am wondering: why do we need to define the full-vector if we are not building for a bootloader, but for a chain-loadable firwmare? That firmware will not use this vector, right? (Well, only the MSP and Reset words are needed).
This dates back to #4536, but can't find the reasoning.

Reply:
> @anangl Is interrupts number available via DT somehow?

There is `NUM_IRQS` in Kconfig. It could be used here.

Reply:
`NUM_IRQS` helped here. Thank you @anangl  for suggestion.

Reply:
@ioannisg Indeed, the last binary in the chain does not need the relay vector table. The `relocate_vector_table` in `aarch32/prep_c.c` would have to do nothing in this case.

Reply:
@e-rk  looks like chain-loaded application has build in __vector_relay_handler referenced in __vector_relay_table build which is not used. Am I right?

Reply:
no @e-rk, I don't think this is correct. The chained binary still needs to set the vector start on the  ram address (normally on 0x20000000), for the bootloader to forward interrupts. But it does not need the relay table.

Reply:
Sorry I phrased it badly. I meant that the last binary *should* not need the vector relay handler, but that's not the case as it is now. Should I address this in this PR?

Reply:
No need, that's only an enhancement to save some space, but might require Kconfig API changes, so I suggest to ignore for now and get this PR fix only the extension of SW IRQ forwarding in Mainline Cortex-M